promotion/datastore.go

```diff
@@ -1696,6 +1849,44 @@ func (pg *Postgres) RunNextDrainRetryJob(ctx context.Context, worker DrainRetryW
 	}
 }
 
+// RunNextFetchRewardGrantsJob - runs fetch reward grant jobs
+func (pg *Postgres) RunNextFetchRewardGrantsJob(ctx context.Context, worker SwapRewardsWorker) error {
+
+	service, _ := InitService(ctx, pg, nil, nil)
+
+	grant, msg, err := worker.FetchRewardsGrants(ctx)
+	if err != nil {
+		return fmt.Errorf("fetch rewards grants job: failed to retrieve grant information: %w", err)
+	}
+
+	cohort := "control"
+	//maybe this call should be changed to just rely on the right status codes and only do 1 post request
+	_, err = service.GetOrCreateIssuer(ctx, grant.PromotionID, cohort)
+	if err != nil {
+		return err
+	}
+
+	claimType := "swap"
```

Comment:
We should use a different (probably new) column to track this information.  This column represents wether the grant is a virtual grant or a direct deposit (meaning we paid directly to the uses Uphold / Gemini accounts).  Swap rewards will all be virtual grants, so the claim_type in your insert should be `'vg'`.

Reply:
I'm fine with this. @husobee any comments?